mdp/test/test_online_flows_hinet.py

```diff
@@ -0,0 +1,305 @@
+           (flow[1] == new_nodes_list[1]), '__setitem__ did not set slice'
+    # test__?etitem__, extended slice
+    flowslice = flow[:2:1]
+    assert isinstance(flowslice,mdp.OnlineFlow), \
+           '__getitem__ slice is not a Flow instance'
+    assert len(flowslice) == 2, '__getitem__ returned wrong slice size'
+    new_nodes_list = [BogusOnlineNode(), BogusOnlineNode()]
+    flow[:2:1] = new_nodes_list
+    assert (flow[0] == new_nodes_list[0]) and \
+           (flow[1] == new_nodes_list[1]), '__setitem__ did not set slice'
+    # test __delitem__, integer key
+    copy_flow = mdp.OnlineFlow(flow[:])
+    del copy_flow[0]
+    assert len(copy_flow) == len(flow)-1, '__delitem__ did not del'
+    for i in range(len(copy_flow)):
+        assert copy_flow[i] == flow[i+1], '__delitem__ deleted wrong node'
+    # test __delitem__, normal slice
+    copy_flow = mdp.OnlineFlow(flow[:])
+    del copy_flow[:2]
+    assert len(copy_flow) == len(flow)-2, \
+           '__delitem__ did not del normal slice'
+    assert copy_flow[0] == flow[2], \
+           '__delitem__ deleted wrong normal slice'
+    # test __delitem__, extended slice
+    copy_flow = mdp.OnlineFlow(flow[:])
+    del copy_flow[:2:1]
+    assert len(copy_flow) == len(flow)-2, \
+           '__delitem__ did not del extended slice'
+    assert copy_flow[0] == flow[2], \
+           '__delitem__ deleted wrong extended slice'
+    # test __add__
+    newflow = flow + flow
+    assert len(newflow) == len(flow)*2, '__add__ did not work'
+
+    # test __add__ with Node
+    flow = _get_default_flow()
+    newflow = flow + BogusNode()
+    assert len(newflow) == len(flow)+1, '__add__ did not work'
+
+    # test insert with Node
+    flow[1] = BogusNode()
+    inp = numx.ones((1,3))*2
+    flow.train(inp)
+    out = flow(inp)
+    rec = flow.inverse(out)
+    assert_array_equal(rec,inp)
+
+    # test __add__ with TrainableNode
+    flow = _get_default_flow()
+    newflow = flow + BogusNodeTrainable()
+    assert len(newflow) == len(flow) + 1, '__add__ did not work'
+
+    # test insert with TrainableNode
+    try:
+        flow[1] = BogusNodeTrainable()
+    except TypeError: pass
+    else: raise Exception('This is not supposed to work!')
+
+
+def test_flow_container_listmethods():
+    # for all methods try using a node with right dimensionality
+    # and one with wrong dimensionality
+    flow = _get_default_flow()
+    length = len(flow)
+    # we test __contains__ and __iter__ with the for loop
+    for node in flow:
+        node.input_dim = 10
+        node.output_dim = 10
+    # append
+    newnode = BogusNode(input_dim=10, output_dim=10)
+    flow.append(newnode)
+    assert_equal(len(flow), length+1)
+    length = len(flow)
+    try:
+        newnode = BogusNode(input_dim=11)
+        flow.append(newnode)
+        raise Exception('flow.append appended inconsistent node')
+    except ValueError:
+        assert_equal(len(flow), length)
+    # extend
```

Comment:
I simply added this test from 'test_flows.py' with modifications to include online flow. In any case, here each list method test uses common values. So, splitting them would require rewriting all the tests.